core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/colsOfKind.kt

```diff
@@ -0,0 +1,86 @@
+package org.jetbrains.kotlinx.dataframe.api
+
+import org.jetbrains.kotlinx.dataframe.ColumnFilter
+import org.jetbrains.kotlinx.dataframe.DataRow
+import org.jetbrains.kotlinx.dataframe.columns.ColumnKind
+import org.jetbrains.kotlinx.dataframe.columns.ColumnPath
+import org.jetbrains.kotlinx.dataframe.columns.ColumnSet
+import org.jetbrains.kotlinx.dataframe.columns.ColumnsResolver
+import org.jetbrains.kotlinx.dataframe.columns.SingleColumn
+import org.jetbrains.kotlinx.dataframe.impl.columns.TransformableColumnSet
+import org.jetbrains.kotlinx.dataframe.impl.headPlusArray
+import kotlin.reflect.KProperty
+
+// region ColumnsSelectionDsl
+
+// TODO tbd
+public interface ColsOfKindColumnsSelectionDsl {
+
+    /** TODO tbd */
+    public fun ColumnSet<*>.colsOfKind(
```

Comment:
is Kind and Child the same abstraction? I suggest descendant instead of Child/Kind/Parent terminology

Reply:
Kind != Child, we're not talking Dutch or German, haha. Kind is the type of column, so either Frame, Value, or ColumnGroup.

Reply:
Also, not part of the current review cycle yet